api/v1alpha1/merge_test.go

```diff
@@ -398,6 +400,244 @@ var _ = Describe("Merge", func() {
+						Spec: &operatorv1alpha1.IstioOperatorSpec{},
+					}
+
+					istioCR := Istio{Spec: IstioSpec{Components: &Components{
+						Cni: &CniComponent{K8S: &CniK8sConfig{
+							Affinity: &v1.Affinity{
+								NodeAffinity: &v1.NodeAffinity{
+									RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+										NodeSelectorTerms: []v1.NodeSelectorTerm{
+											{
+												MatchExpressions: []v1.NodeSelectorRequirement{
+													{
+														Key:      "app-new",
+														Operator: "In",
+														Values:   Label("istio-cni-node1"),
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						}},
+					}}}
+
+					// when
+					out, err := istioCR.MergeInto(iop)
+
+					// then
+					Expect(err).ShouldNot(HaveOccurred())
+
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms).To(HaveLen(1))
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions).To(HaveLen(1))
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Key).To(Equal("app-new"))
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Operator).To(BeEquivalentTo("In"))
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values).To(HaveLen(1))
+					Expect(out.Spec.Components.Cni.K8S.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values[0]).To(BeEquivalentTo("istio-cni-node1"))
+				})
+			})
+		})
+
+		Context("Resources", func() {
+			It("Should update CNI resources when those are present in Istio CR", func() {
+				//given
+				iop := istioOperator.IstioOperator{
+					Spec: &operatorv1alpha1.IstioOperatorSpec{},
+				}
+				cpuRequests := "500m"
+				memoryRequests := "500Mi"
+
+				istioCR := Istio{Spec: IstioSpec{Components: &Components{
+					Cni: &CniComponent{K8S: &CniK8sConfig{
+						Resources: &Resources{
+							Requests: &ResourceClaims{
+								Cpu:    &cpuRequests,
+								Memory: &memoryRequests,
+							},
+						},
+					}},
+				}}}
+
+				// when
+				out, err := istioCR.MergeInto(iop)
+
+				// then
+				Expect(err).ShouldNot(HaveOccurred())
+
+				iopCpuRequests := out.Spec.Components.Cni.K8S.Resources.Requests["cpu"]
+				Expect(iopCpuRequests).To(Equal(cpuRequests))
+
+				iopMemoryRequests := out.Spec.Components.Cni.K8S.Resources.Requests["memory"]
+				Expect(iopMemoryRequests).To(Equal(memoryRequests))
+			})
+		})
+	})
+
+	Context("Proxy", func() {
+		It("Should update Proxy resources configuration if they are present in Istio CR", func() {
```

Comment:
This test has exactly the same name as the following Proxy test.
Do we need this test, since the other test also sets the requests?

Reply:
We don't :)